app/Http/Controllers/EventRegistrationController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\RegistrationStoreRequest;
+use App\Models\Character;
+use App\Models\Event;
+use App\Models\Session;
+use Illuminate\Http\Request;
+use Inertia\Inertia;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+
+class EventRegistrationController extends Controller
+{
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function store(RegistrationStoreRequest $request)
+    {
+        // this could probably be a request validator...
+        $data = $request->validated();
+
+        $character = Character::find($data['character_id']);
+        $request->user()->can('update', $character);
```

Comment:
Shouldn't this be in the request authorize method?

Reply:
Yeah you're right i just didn't want to be doubling up on the character fetch request, all good though shouldn't hurt performance in any tangible way, Fixed!